bwresources.py

```diff
@@ -173,23 +173,19 @@ def get_mentions(self, max_pages=None, **kwargs):
         """
         params = self._fill_mentions_params(kwargs)
         all_mentions = []
-        counter = 0
-        next_end_date = params["endDate"]
 
-        while max_pages == None or counter < max_pages:
-            params["endDate"] = next_end_date
-            next_mentions = self.get_mentions_page(params, 0)
+        while max_pages == None or params["pageSize"] < max_pages:
+            next_mentions = self._get_mentions_page(params, params["pageSize"])
```

Comment:
I think this should be `params["page"]` rather than `params["pageSize"]`.  The `pageSize` parameter in the API determines how many mentions you get per request, while the `page` parameter determines which block of mentions you get back.  (For instance to get 15000 mentions you could make 3 requests each with `pageSize` 5000 and with `page` equal to 0 in the first request, 1 in the second, and 2 in the third.)  I added a commit to this branch which does things the way I think it should work (and confirmed with a test).